test_scripts/API/ButtonSubscription/commonButtonSubscription.lua

```diff
@@ -0,0 +1,438 @@
+  :Do(function(_, data)
+      m.getHMIConnection():SendResponse(data.id, "BasicCommunication.ActivateApp", "SUCCESS", {})
+    end)
+end
+
+--[[ @ignitionOff: perform ignition off
+--! @parameters: none
+--! @return: none
+--]]
+function m.ignitionOff()
+  local isOnSDLCloseSent = false
+  m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "SUSPEND" })
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLPersistenceComplete")
+  :Do(function()
+    m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "IGNITION_OFF" })
+    m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLClose")
+    :Do(function()
+      isOnSDLCloseSent = true
+      SDL.DeleteFile()
+    end)
+  end)
+  m.wait(3000)
+  :Do(function()
+    if isOnSDLCloseSent == false then m.cprint(color.magenta, "BC.OnSDLClose was not sent") end
+    for i = 1, actions.mobile.getAppsCount() do
+      actions.mobile.deleteSession(i)
+    end
+    StopSDL()
+  end)
+end
+
+--[[ @reRegisterApp: re-register application with RESUME_FAILED resultCode
+--! @parameters:
+--! pAppId: application number (1, 2, etc.)
+--! pCheckResumptionData - verification function for resumption data
+--! pRAIResponseExp - time for expectation of RAI response
+--! @return: none
+--]]
+function m.reRegisterApp(pAppId, pCheckResumptionData, pRAIResponseExp)
+  if not pAppId then pAppId = 1 end
+  if not pRAIResponseExp then pRAIResponseExp = 10000 end
+  local mobSession = m.getMobileSession(pAppId)
+  mobSession:StartService(7)
+  :Do(function()
+      local params = m.cloneTable(m.getConfigAppParams(pAppId))
+      params.hashID = m.hashId[pAppId]
+      local corId = mobSession:SendRPC("RegisterAppInterface", params)
+      m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered",
+        { application = { appName = m.getConfigAppParams(pAppId).appName } })
+      mobSession:ExpectResponse(corId, { success = true, resultCode = "RESUME_FAILED" })
+      :Do(function()
+          m.getHMIConnection():ExpectRequest("BasicCommunication.ActivateApp", { appID = m.getHMIAppId(pAppId) })
+          :Do(function(_, data)
+              m.getHMIConnection():SendResponse(data.id, "BasicCommunication.ActivateApp", "SUCCESS", {})
+            end)
+          mobSession:ExpectNotification("OnPermissionsChange")
+        end)
+      :Timeout(pRAIResponseExp)
+    end)
+  pCheckResumptionData(pAppId)
+end
+
+--[[ @removeButtonFromCapabilities: remove button from HMI capabilities
+--! @parameters:
+--! pButtonName - button name
+--! @return: returns HMI Table without button
+--]]
+function m.removeButtonFromCapabilities(pButtonName)
+  local hmiValues = m.getDefaultHMITable()
+  for i, buttonNameTab in pairs(hmiValues.Buttons.GetCapabilities.params.capabilities) do
+    if (buttonNameTab.name == pButtonName) then
+      table.remove(hmiValues.Buttons.GetCapabilities.params.capabilities, i)
```

Comment:
```suggestion
      table.remove(hmiValues.Buttons.GetCapabilities.params.capabilities, i)
      return hmiValues
```
can we short circuit this?

Reply:
@iCollin Definitely! Please find update in a476d04